### PR TITLE
cpu-flex: in-order/in-order-superscalar parameters

### PR DESCRIPTION
**Pull request by animalvgamer**
Change-Id: I5a557a668300efae70b2ed8a839b6a5df62247ab
Signed-off-by: Bradley Wang <radwang@ucdavis.edu>

### CODE REVIEW

File: src/cpu/flexcpu/flexcpu_thread.cc [hunk 1]
```diff
@@ -1001,13 +1006,21 @@ FlexCPUThread::populateDependencies(shared_ptr<InflightInst> inst_ptr)
...
+        const shared_ptr<InflightInst>& last_inst =
+                                                *(++inflightInsts.rbegin());
 
-    //     if (!(last_inst->isComplete() || last_inst->isSquashed()))
-    //         inst_ptr->addDependency(last_inst);
-    // }
+        if (!(last_inst->isComplete() || last_inst->isSquashed()))
+            inst_ptr->addDependency(*last_inst);
+    }
+
```
Comment (powerjg) [inline src/cpu/flexcpu/flexcpu_thread.cc]: Add a comment here describing this?

File: src/cpu/flexcpu/flexcpu_thread.cc [hunk 2]
```diff
@@ -1001,13 +1006,21 @@ FlexCPUThread::populateDependencies(shared_ptr<InflightInst> inst_ptr)
     //       interface, to make these rules swappable?
 
     // Temporary virtual dependency rule to simulate in-order CPU mode.
```
Comment (powerjg) [inline src/cpu/flexcpu/flexcpu_thread.cc]: Expand this comment to explain what the dependency *is*?

File: src/cpu/flexcpu/flexcpu_thread.cc (hunk 1 above)
Comment (animalvgamer) [reply src/cpu/flexcpu/flexcpu_thread.cc]: Done

File: src/cpu/flexcpu/flexcpu_thread.cc (hunk 2 above)
Comment (animalvgamer) [reply src/cpu/flexcpu/flexcpu_thread.cc]: done

**Pull request merged**